keel-sql/src/main/resources/db/changelog/202010303-baked-images.yml

```diff
@@ -22,32 +27,27 @@ databaseChangeLog:
         - sql:
             sql: |
               alter table baked_images
-              add column package_version varchar(150) generated always as (image ->> '$.packageVersion')
+              add column package_version varchar(150) generated always as (image ->> '$.appVersion')
               not null
               first;
         - sql:
             sql: |
               alter table baked_images
-              add column cloud_provider varchar(150) generated always as (image ->> '$.cloudProvider')
+              add column cloud_provider varchar(50) generated always as (image ->> '$.cloudProvider')
               not null
               after package_version;
         - sql:
             sql: |
               alter table baked_images
-              add column base_label varchar(150) generated always as (image ->> '$.baseLabel')
+              add column base_label varchar(50) generated always as (image ->> '$.baseLabel')
               not null
               after cloud_provider;
         - sql:
             sql: |
               alter table baked_images
-              add column base_os varchar(150) generated always as (image ->> '$.baseOs')
+              add column base_os varchar(50) generated always as (image ->> '$.baseOs')
               not null
               after base_label;
-        - sql:
-            sql: |
-              alter table baked_images
-              add column time_detected datetime(3) generated always as (str_to_date(image->>'$.timestamp', '%Y-%m-%dT%T.%fZ'))
```

Comment:
Did you maybe have left-over data in your local database generated when you only had `timestamp` (smaller precision), which would break the `datetime(3)` here? Just a guess, but I remember @robfletcher had to add the missing precision in the migration because we had a few records like in the DB.

Reply:
No, I had to wipe my local database everytime I changed the sql. I didn't feel like it was worth figuring out why I couldn't get this fancy bit to work, discussed with rob offline.